src/diet_journal.py

```diff
@@ -0,0 +1,228 @@
+from datetime import datetime, timedelta
+from enum import Enum
+import json
+
+from postgres import with_postgres_connection
+
+
+class NoValue(Enum):
+    def __repr__(self):
+        return '%s' % self.value
+
+
+class Food:
+    def __init__(self, name="", calories=0):
+        self.name = name
+        self.calories = calories
+
+
+class MealType(str, NoValue):
+    BREAKFAST = 'Breakfast'
+    LUNCH = 'Lunch'
+    DINNER = 'Dinner'
+    SNACK = 'Snack'
+    MIDNIGHTSNACK = 'Midnight Snack'
+
+
+class BodyType(str, NoValue):
+    HIPS = 'Hips'
+    WAIST = 'Waist'
+    BUST = 'Bust'
+    LEFTARM = 'Left Arm'
+    RIGHTARM = 'Right Arm'
+    LEFTTHIGH = 'Left Thigh'
+    RIGHTTHIGH = 'Right Thigh'
+    NECK = 'Neck'
+
+
+class JournalEntry():
+    def __init__(self):
+        self.entries_by_meal_type = {
+            MealType.BREAKFAST: [],
+            MealType.LUNCH: [],
+            MealType.DINNER: [],
+            MealType.SNACK: [],
+            MealType.MIDNIGHTSNACK: [],
+        }
+
+    def update_meal(self, meal_type, food_entries):
+        self.entries_by_meal_type[meal_type].extend(food_entries)
+
+    def to_json(self):
+        return json.dumps(self.entries_by_meal_type, default=lambda o: o.__dict__, indent=4)
```

Comment:
JSOn store forces u to do this. `indent=4` is using up storage for whitespace, btw